### ISSUE TITLE
ghost, ghost_pull and custom_mapper fails on ppc64

### ISSUE THREAD

**Issue by junghans** [StanfordLegion/legion#233]
After applying #229 and #232, some test still fail on ppc64.

Detail for `ghost`:
```
      Start  4: ghost
4: Test command: /builddir/build/BUILD/legion-legion-17.02.0/serial/bin/ghost "-ll:cpu" "4"
4: Test timeout computed to be: 9.99988e+06
4: ghost: /builddir/build/BUILD/legion-legion-17.02.0/runtime/realm/runtime_impl.cc:1704: Realm::GenEventImpl* Realm::RuntimeImpl::get_genevent_impl(Realm::Event): Assertion `check.event.creator_node == i
d.event.creator_node' failed.
4: can't read '/sys/devices/system/cpu/cpu-1/topology/core_id' - skippingcan't read '/sys/devices/system/cpu/cpu-1/topology/core_id' - skippingcan't read '/sys/devices/system/cpu/cpu-1/topology/core_id' -
 skippingcan't read '/sys/devices/system/cpu/cpu-1/topology/core_id' - skipping[0 - 3fff8b0084d0] {4}{threads}: reservation ('DMA request queue') cannot be satisfied
4: Running stencil computation for 1024 elements for 10 steps...
4: Partitioning data into 4 sub-regions...
 4/18 Test  #4: ghost ............................***Exception: Other  0.24 sec
```
and for `ghost_pull`:
```
     Start  5: ghost_pull
5: Test command: /builddir/build/BUILD/legion-legion-17.02.0/serial/bin/ghost_pull "-ll:cpu" "4"
5: Test timeout computed to be: 9.99988e+06
5: ghost_pull: /builddir/build/BUILD/legion-legion-17.02.0/runtime/realm/runtime_impl.cc:1704: Realm::GenEventImpl* Realm::RuntimeImpl::get_genevent_impl(Realm::Event): Assertion `check.event.creator_node
 == id.event.creator_node' failed.
5: can't read '/sys/devices/system/cpu/cpu-1/topology/core_id' - skippingcan't read '/sys/devices/system/cpu/cpu-1/topology/core_id' - skippingcan't read '/sys/devices/system/cpu/cpu-1/topology/core_id' -
 skippingcan't read '/sys/devices/system/cpu/cpu-1/topology/core_id' - skipping[0 - 3fff943d84d0] {4}{threads}: reservation ('DMA request queue') cannot be satisfied
5: Running stencil computation for 1024 elements for 10 steps...
5: Partitioning data into 4 sub-regions...
 5/18 Test  #5: ghost_pull .......................***Exception: Other  0.34 sec
```
and `custom_mapper`:
```
      Start 17: custom_mapper
17: Test command: /builddir/build/BUILD/legion-legion-17.02.0/serial/bin/custom_mapper
17: Test timeout computed to be: 9.99988e+06
17: can't read '/sys/devices/system/cpu/cpu-1/topology/core_id' - skippingcan't read '/sys/devices/system/cpu/cpu-1/topology/core_id' - skippingcan't read '/sys/devices/system/cpu/cpu-1/topology/core_id' - skippingcan't read '/sys/devices/system/cpu/cpu-1/topology/core_id' - skippingRunning daxpy for 1024 elements...
17: Partitioning data into 0 sub-regions...
17: [0 - 3fff6d64f130] {5}{runtime}: Attempt to create index partition with no colors in task unnamed_task_0 (ID 1). Index partitions must have at least one color.
17/18 Test #17: custom_mapper ....................***Failed    0.01 sec
```

Build details in ([build.log.txt](https://github.com/StanfordLegion/legion/files/853457/build.log.txt) and [here](https://koji.fedoraproject.org/koji/taskinfo?taskID=18476246).

**Comment by junghans** [comment]
With 877c07f29141ee94f0b0fb076eb6d32de844b935 it is back to segfault: [build.log.txt](https://github.com/StanfordLegion/legion/files/856018/build.log.txt)

**Comment by streichler** [comment]
@junghans Did this start passing at some point before 877c07f?

**Comment by junghans** [comment]
It never passed, but looking at the error output before 877c07f29141ee94f0b0fb076eb6d32de844b935, it seems like hwloc return `cpu_id = -1` on ppc64.

**Comment by junghans** [comment]
This problem still persist with legion-17.05.0, details here: [build.log](https://github.com/StanfordLegion/legion/files/1033201/build.log.txt)

**Comment by junghans** [comment]
Seems to be fixed in legion-17.08.0
